woocommerce/wc-admin
# Implement the PayU payment task and fix a bug

Fixes #6172 

This adds the PayU plugin as a payment task option. Based on the requirements, we've opted for this just to be a simple task that installs the plugin and lets the user configure the plugin later via it's settings screen.

There was also an undiscovered bug which I reproduced with Klarna and Mollie, which is that when you click continue on the payment task calling `markConfigured` it doesn't persist the enabling of the plugin, so a page reload shows the toggle disabled but also the payment plugin will not be enabled either.

To fix this, I've mimicked the behaviour from the toggle functionality to persist the actual enabled option as well as setting the state.

### Detailed test instructions:

To test the basic plugin functionality:

1. Go through the OBW and choose India as country, the only other thing that matters is to not have cbd as industry
2. After landing back on home screen, click the setup payment task in the task list
3. You should see PayU listed in the possible options, click the "setup" button for PayU
4. The payu-india plugin should be installed automatically, in the next step instructions matching the copy from the issue should be displayed. After clicking "continue" it should return to the payment plugin list and the toggle should be enabled
5. Do a page refresh, the plugin option should still be enabled.

Test some of the other plugins in the same way to ensure that they are persisted as enabled. Some examples include Klarna and Mollie. For Klarna and Mollie you'll need to change the country in OBW to something like "United Kingdom".
